src/core/InitiativeOrder/InitiativeOrder.ts

```diff
@@ -0,0 +1,231 @@
+import {
+  InitiativeItem,
+  InitiativeOrderState
+} from '@core/types';
+
+export interface InitiativeOrderInterface {
+  getCurrentId(): string;
```

Comment:
Why is it current ID? Does that change?

Reply:
Yup, this is a weird case that came up the last time I tinkered with initiative code. If you use an array to manage initiative, you don't want to store the current player as an array index. This is because adding new creatures to initiative and sorting after that loses that player's current turn order. Using an id ensures that the current turn preserves the correct player.

Using a linked list is an option and storing that node, but that had its own issues last time I tried that.

With array index:

1. Add all players to turn order and first wave of creatures
2. Combat goes on
3. Add a new wave of creatures with new initiative rolls
4. Re-sort
5.  Current player's turn is swapped with whatever creature fills array[x]